kornia/filters/gaussian.py

```diff
@@ -91,6 +95,14 @@ def __init__(self,
         self.sigma: Tuple[float, float] = sigma
```

Comment:
maybe we could have the `kernel_size` as private member and access by the property decorator. Then have a setter so that we generate and update the kernels in case the user sets a new kernel size.

Reply:
I've updated the three properties kernel_size, sigma, and separable to be accessed through property decorator, and they will recompute the kernel when set. There's also a bit of extra logic to delete extraneous buffers when switching between separable and non-separable.